client/src/components/DBModal.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react';
+import apiClient from '../lib/ApiClient';
+
+const DBModal = ({ modalOpen, setModalOpen, setDbName }) => {
+  const [user, setUser] = useState("");
+  const [host, setHost] = useState("");
+  const [password, setPassword] = useState("");
+  const [database, setDatabase] = useState("");
+  const [port, setPort] = useState("");
+
+  const handleCloseModal = () => {
+    setModalOpen(false);
+  };
+
+  const validateForm = (dbObj) => {
+    if (Object.values(dbObj).some(val => val === "")) {
+      return "All fields must be filled in."
+    }
+
+    if (!/^[a-z\d.-]+$/i.test(dbObj.host)) {
+      return "Invalid host name";
+    }
+
+    if (Number(dbObj.port) < 0 || Number(dbObj.port) > 65535) {
+      return "Invalid port number. Must be a number between 0 and 65535"
+    }
+  }
+
+  const resetForm = () => {
+    setModalOpen(false);
+    setUser("")
+    setHost("")
+    setPassword("")
+    setDatabase("")
+    setPort("")
+  }
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const dbObj = { user, host, password, database, port };
+    const invalidPropMessage = validateForm(dbObj);
+    if (invalidPropMessage) {
+      alert(invalidPropMessage);
+      return
+    }
+
+    apiClient.connectToDB(dbObj, () => {
+      setDbName(database);
+      resetForm();
+    })
+  }
+
+  return (
+    <div className={`overlay ${modalOpen ? "" : "hidden"}`}>
+      <div className="new-flag-modal">
+        <i className="x-icon icon close-modal" onClick={handleCloseModal}></i>
+        <h2 className="font-bold text-xl text-primary-violet">Database Connection</h2>
+        <form className="database-form">
+          <div className="mt-2.5">
+            <label htmlFor="user" className="mr-5">Username: </label>
+            <input id="user" type="text" className="border border-primary-oxfordblue rounded-lg px-2" value={user} onChange={(e) => setUser(e.target.value)} />
+          </div>
+          <div className="mt-2.5">
+            <label htmlFor="host" className="mr-5">Host: </label>
+            <input id="host" type="text" className="border border-primary-oxfordblue rounded-lg px-2" value={host} onChange={(e) => setHost(e.target.value)} />
+          </div>
+          <div className="mt-2.5">
+            <label htmlFor="password" className="mr-5">Password: </label>
+            <input id="password" type="text" className="border border-primary-oxfordblue rounded-lg px-2" value={password} onChange={(e) => setPassword(e.target.value)} />
+          </div>
+          <div className="mt-2.5">
+            <label htmlFor="database" className="mr-5">Database: </label>
+            <input id="database" type="text" className="border border-primary-oxfordblue rounded-lg px-2" value={database} onChange={(e) => setDatabase(e.target.value)} />
+          </div>
+          <div className="mt-2.5" className="mr-5">
```

Comment:
This div has `className` twice, which is breaking the CSS for this div. I think you meant to put the second className on the `label` element

Reply:
Good catch! Just fixed it.